service/src/scripts/register-profile.ts

```diff
@@ -0,0 +1,56 @@
+import 'dotenv/config'
+import { getWeb3Interface } from './web3-interface'
+import { validateEnvVars } from '../utils/validateEnv'
+import { exit } from 'process'
+import { RPC_EXCEPTION } from '../utils/exit-codes'
+import { archLogger } from '../utils/chalk-theme'
+import { parseRegisterArgs } from '../utils/cli_parsers/parseRegisterArgs'
+import { getOnchainProfile } from '../utils/onchain-data'
+import { ethers } from 'ethers'
+import { requestApproval } from './approve_utils'
+
+validateEnvVars();
+
+const web3Interface = await getWeb3Interface();
+const profile = await getOnchainProfile(web3Interface);
+
+archLogger.notice("Registering your Archaeologist profile...");
+
+if (profile.exists) {
+  archLogger.notice("Already registered!");
+  exit(0);
+}
+
+
+const handleException = (e) => {
+  archLogger.error(e);
+  exit(RPC_EXCEPTION);
+}
+
+const { minimumDiggingFee, maximumRewrapInterval, freeBond } = await parseRegisterArgs(web3Interface);
+
+let freeBondDeposit = ethers.constants.Zero;
+
+if (freeBond.gt(ethers.constants.Zero)) {
+  const approved = await requestApproval(
+    web3Interface,
+    "You will need to approve Sarcophagus contracts to use your SARCO in order to deposit free bond.\nEnter 'approve' to authorize this, or else hit <ENTER> to continue without a deposit:",
```

Comment:
The intention here - especially since this is a "do-once-and-never-again thing" - was to emulate the intentional nature of approving via a wallet, or actually typing out `npm run approve` with full knowledge of what the user is opting in for, as opposed to accidentally hitting `ENTER` and giving said approval.

Of course a max-int approval seems standard practice, but in the spirit of being accessible to people who aren't necessarily experienced with interacting with contracts, I'd argue that accidentally saying 'no' to this prompt is actually better UX.